dbms/src/Columns/ColumnArray.h

```diff
@@ -108,9 +155,24 @@ class ColumnArray final : public COWPtrHelper<IColumn, ColumnArray>
             ErrorCodes::NOT_IMPLEMENTED);
     }
 
-    void deserializeAndInsertFromPos(PaddedPODArray<char *> & pos, bool use_nt_align_buffer) override;
+    void deserializeAndInsertFromPosUnique(
+        PaddedPODArray<const char *> & pos,
+        bool use_nt_align_buffer,
+        const TiDB::TiDBCollatorPtr & collator) override;
+    void deserializeAndInsertFromPos(PaddedPODArray<const char *> & pos, bool use_nt_align_buffer) override;
+
+    void deserializeAndInsertFromPosUniqueForColumnArray(
+        PaddedPODArray<const char *> & /* pos */,
+        const IColumn::Offsets & /* array_offsets */,
+        bool /* use_nt_align_buffer */,
+        const TiDB::TiDBCollatorPtr & /* collator */) override
+    {
+        throw Exception(
+            "Method deserializeAndInsertFromPosUniqueForColumnArray is not supported for " + getName(),
+            ErrorCodes::NOT_IMPLEMENTED);
+    }
     void deserializeAndInsertFromPosForColumnArray(
-        PaddedPODArray<char *> & /* pos */,
+        PaddedPODArray<const char *> & /* pos */,
```

Comment:
The `PaddedPODArray<char *>` variable can not pass to `PaddedPODArray<const char *>` so it's better not to change it.

Reply:
ok